drivers/include/periph/rtc.h

```diff
@@ -85,8 +89,10 @@ int rtc_set_alarm(struct tm *time, rtc_alarm_cb_t cb, void *arg);
  *
  * @param[out]  time        Pointer to structure to receive alarm time
  *
- * @return  0 for success
- * @return -1 an error occurred
+ * @retval  0               Success
+ * @retval -EINVAL          @p time is `NULL`
+ * @retval -ENODATA         No alarm is pending
```

Comment:
No error code fully matches this. These would be the closes matches:

| Error Code | Output of `strerror(errno)`                           |
|:---------- |:----------------------------------------------------- |
| `ETIME`    | Timer expired                                         |
| `ENODATA`  | No message is available on the STREAM head read queue |
| `ENOENT`   | No such file or directory                             |
| `ENOMSG`   | No message of the desired type                        |

All of these are pretty poor choices when looking at the error code description, but for `ENODATA` at least the name of the error code is somewhat fitting.

Reply:
I would prefere `ENODATA`.